analyzers/tests/SonarAnalyzer.UnitTest/SymbolicExecution/Roslyn/RoslynSymbolicExecutionTest.Invocation.cs

```diff
@@ -962,6 +968,65 @@ public void Invocation_Equals_SplitsToBothResults(string left, string right)
                 });
         }
 
+        [DataTestMethod]
+        [DataRow("null", "null", true)]
+        [DataRow("null", "42", false)]
+        [DataRow("42", "null", false)]
+        public void Invocation_NullableEquals_LearnResult(string left, string right, bool expectedResult)
+        {
+            var code = $"""
+                int? left = {left};
+                int? right = {right};
+                var result = left.Equals(right);
+                Tag("Result", result);
+                """;
+            SETestContext.CreateCS(code).Validator.ValidateTag("Result", x => x.Should().HaveOnlyConstraints(ObjectConstraint.NotNull, BoolConstraint.From(expectedResult)));
+        }
+
+        [DataTestMethod]
+        [DataRow("42", "42")]
+        [DataRow("42", "0")]
+        [DataRow("0", "42")]
+        [DataRow("42", "Unknown<int?>()")]
+        [DataRow("Unknown<int?>()", "42")]
+        [DataRow("Unknown<int?>()", "Unknown<int?>()")]
+        public void Invocation_NullableEquals_DoesNotLearnResult(string left, string right)
```

Comment:
Can we have the same for BoolConstraint? I would expect some "result" to have a "true" or "false" constraint sometime with branching (like in `Invocation_ObjectEquals_SplitsToBothResults`) and sometimes without (e.g. if left and right both have a BoolConstraint, or one has a BoolConstraint and the other has NullConstraint).